tests/auto/tst_declarativetabmodel/tst_declarativetabmodel.cpp

```diff
@@ -1,493 +0,0 @@
-    int tab2 = currentTabId();
-    QVERIFY(tab1 != tab2);
-    QCOMPARE(tabModel->activeTab().url(), page1Tab2Url);
-    QCOMPARE(tabModel->activeTab().title(), page1Tab2Title);
-    QTest::qWait(1000);
-
-    int index = tabModel->findTabIndex(tab1);
-    QModelIndex modelIndex = tabModel->createIndex(index, 0);
-
-    // tab1 has page2Tab1Url and empty title still.
-    QCOMPARE(tabModel->data(modelIndex, DeclarativeTabModel::UrlRole).toString(), page2Tab1Url);
-    QVERIFY(tabModel->data(modelIndex, DeclarativeTabModel::TitleRole).toString().isEmpty());
-
-    QTest::qWait(1000);
-
-    QString page2Tab2Url = page2Tab1Url;
-    QString page2Tab2Title = "Second Page Too";
-    tabModel->updateUrl(tab2, page2Tab2Url, false);
-
-    QTest::qWait(1000);
-
-    QCOMPARE(tabModel->activeTab().url(), page2Tab2Url);
-    QVERIFY(tabModel->activeTab().title().isEmpty());
-
-    QCOMPARE(tabModel->data(modelIndex, DeclarativeTabModel::UrlRole).toString(), page2Tab1Url);
-    QVERIFY(tabModel->data(modelIndex, DeclarativeTabModel::TitleRole).toString().isEmpty());
-
-    QTest::qWait(1000);
-
-    QSignalSpy activeTabChangedSpy(tabModel, SIGNAL(activeTabChanged(int,bool)));
-    int expectedCount = tabModel->count() - 2;
-    tabModel->removeTabById(tab1, true);
-    waitSignals(activeTabChangedSpy, 1);
-    tabModel->removeTabById(tab2, true);
-    waitSignals(activeTabChangedSpy, 2);
-    QVERIFY(tabModel->count() == expectedCount);
-}
-
-void tst_declarativetabmodel::updateInvalidUrls_data()
-{
-    QString expectedUrl = tabModel->activeTab().url();
-    QTest::addColumn<QString>("url");
-    QTest::addColumn<QString>("expectedUrl");
-    QTest::newRow("tel") << "tel:+123456798" << expectedUrl;
-    QTest::newRow("sms") << "sms:+123456798" << expectedUrl;
-    QTest::newRow("mailto") << "mailto:joe@example.com" << expectedUrl;
-    QTest::newRow("mailto query does not count") << "mailto:joe@example.com?cc=bob@example.com&body=hello1" << expectedUrl;
-    QTest::newRow("geo") << "geo:61.49464,23.77513" << expectedUrl;
-    QTest::newRow("geo://") << "geo://61.49464,23.77513" << expectedUrl;
-}
-
-void tst_declarativetabmodel::updateInvalidUrls()
-{
-    QFETCH(QString, expectedUrl);
-    QFETCH(QString, url);
-    tabModel->updateUrl(currentTabId(), url, false);
-    QTest::qWait(1000);
-    QCOMPARE(tabModel->activeTab().url(), expectedUrl);
-}
-
-void tst_declarativetabmodel::updateValidUrls_data()
-{
-    QTest::addColumn<QString>("url");
-    QTest::newRow("http") << "http://foobar";
-    QTest::newRow("https") << "https://foobar";
-    QTest::newRow("file") << "file://foo/bar/index.html";
-    QTest::newRow("relative") << "foo/bar/index.html";
-}
-
-void tst_declarativetabmodel::updateValidUrls()
-{
-    QFETCH(QString, url);
-
-    int tabId = currentTabId();
-    tabModel->updateUrl(tabId, url, false);
-    QCOMPARE(tabModel->activeTab().url(), url);
-    QTest::qWait(1000);
-}
-
-void tst_declarativetabmodel::invalidTabs_data()
```

Comment:
This could be moved as is to the tst_persistenttabmodel.cpp to test adding tabs with invalid urls. Later that should be extended to take into account also malformed urls like without protocol.